### PR TITLE
Adds additional escaping throughout plugin (VIP)

### PR DESCRIPTION
**Pull request by tomjn**
There's a number of issues flagged during VIP review, I'm currently going through and fixing them by adding in the missing escaping. WIP PR, questions welcome

### CODE REVIEW

File: includes/class-wp-job-manager-ajax.php
Comment (jom) [inline includes/class-wp-job-manager-ajax.php]: As a behavior change, I might cherry-pick this commit to a separate PR, if that works for you? There could be some themes doing things they shouldn't with `-1`.

Comment (tomjn) [reply includes/class-wp-job-manager-ajax.php]: Sounds good

Comment (jom) [reply includes/class-wp-job-manager-ajax.php]: I started doing this and then realized `absint()` prevented `-1` 😉 . Seems redundant.

Comment (tomjn) [reply includes/class-wp-job-manager-ajax.php]: eh but it does prevent passing `0`, I wanted to add a `min` too so there was an upper limit but i figured that would be problematic

Comment (jom) [reply includes/class-wp-job-manager-ajax.php]: In testing, `0` resulted in the default `per_page`, but oh well.

File: templates/account-signin.php
Comment (jom) [inline templates/account-signin.php]: This was an original lint issue, but while editing line, there is a spacing issue in the argument list for `__()`.

**Review by jom — DISMISSED** [review]
Thanks for this PR, @tomjn! Huge improvement. I've added some (mostly linting) notes.

Did some general testing and seems to work well. I've marked it for 1.31.1. I can update the template versions before merging.

**Comment by jom** [comment]
cc: @tripflex I don't _think_ the escaping happening in the `fields/*.php` templates will affect your plugin, but just wanted to flag you.

File: templates/account-signin.php
Comment (tomjn) [reply templates/account-signin.php]: fixed

File: wp-job-manager-template.php
Comment (tomjn) [reply wp-job-manager-template.php]: fixed

File: tests/php/bootstrap.php
Comment (tomjn) [reply tests/php/bootstrap.php]: fixed

File: templates/job-submit.php
Comment (tomjn) [reply templates/job-submit.php]: fixed

File: templates/job-dashboard.php
Comment (tomjn) [reply templates/job-dashboard.php]: Just being more explicit

File: templates/emails/plain/admin-expiring-job.php
Comment (tomjn) [reply templates/emails/plain/admin-expiring-job.php]: I tried opening this file and couldn't see the issue, perhaps I fixed it in a previous commit

File: templates/emails/plain/admin-expiring-job.php
Comment (tomjn) [reply templates/emails/plain/admin-expiring-job.php]: fixed?

File: templates/emails/employer-expiring-job.php
Comment (tomjn) [reply templates/emails/employer-expiring-job.php]: fixed

File: includes/admin/class-wp-job-manager-cpt-legacy.php
Comment (tomjn) [reply includes/admin/class-wp-job-manager-cpt-legacy.php]: fixed

File: includes/admin/class-wp-job-manager-cpt-legacy.php
Comment (tomjn) [reply includes/admin/class-wp-job-manager-cpt-legacy.php]: fixed

File: includes/admin/class-wp-job-manager-cpt.php
Comment (tomjn) [reply includes/admin/class-wp-job-manager-cpt.php]: fixed

File: includes/admin/class-wp-job-manager-writepanels.php
Comment (tomjn) [reply includes/admin/class-wp-job-manager-writepanels.php]: fixed, and I fixed a tonne of other bits that were there from prior

File: includes/admin/class-wp-job-manager-writepanels.php
Comment (tomjn) [reply includes/admin/class-wp-job-manager-writepanels.php]: fixed

File: templates/job-dashboard.php
Comment (jom) [reply templates/job-dashboard.php]: Alright, but `count()` can't return anything but an `int`.

File: assets/js/ajax-filters.js
Comment (jom) [inline assets/js/ajax-filters.js]: `result.showing_links` is contains HTML and gets encoded using `$.text()`. 

screen shot 2018-06-19 at 4 55 28 pm

**Comment by jom** [comment]
I've force pushed a modified version of that commit with #1503's content removed. I want to test that change with WPML before merging and don't want it to hold up this commit. You'll need to do a hard reset to the remote branch.... sorry, since I'm planning on squashing this PR anyway I could have just done a partial revert commit. 

Can we revert the change to line 125 in `ajax-filters.js`? Once we do that, I'll update the template versions for 1.31.1.

**Comment by tomjn** [comment]
@jom my bandwidth at the moment is quite tight, and we've a client waiting on these changes for quite a while now, anything you can do to expedite things would be super helpful

**Comment by jom** [comment]
@tomjn I'll revert the change mentioned in `ajax-filters.js` so the raw HTML doesn't show up (unless you have another preferred method) and then we should be good for this PR.

**Pull request merged**
